### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by bordoley**
This PR updates Immutable.re to use the new Reason workflows for both native and js development. In addition, it adds automation scripts for generating META and opam files using the version specified in package.json. Also added automated deploy to OPAM/NPM via the release script.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,7 +16,12 @@ _build/**
...
 *.log
-/node_modules/
-/lib/
-/immutable.install
-/pkg/META
+node_modules
+immutable.install
+
+# Ignore bsb generated merlin files
+/reUnit/.merlin
```
Comment (cristianoc) [inline .gitignore]: where's reUnit coming from?

File: Makefile [hunk 2]
```diff
@@ -1,29 +1,18 @@
 build:
-	cp pkg/META.in pkg/META
 	ocamlbuild -package topkg pkg/build.native
 	./build.native build
 
-ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
-SUBSTS:=$(ROOT_DIR)/pkg/substs
+test:
+	rebuild -cflag -g -Is src,src/sets,src/core,src/queues,src/indexed,src/maps,src/utils,reUnit/src,test/testers,test ./test/ImmutableTest.byte
```
Comment (cristianoc) [inline Makefile]: perhaps de-dupe common part of build targets

File: .gitignore (hunk 1 above)
Comment (bordoley) [inline .gitignore]: It's a local subfolder. I'm working with the bs-container guys about pulling it into its own external package at some point.

File: reUnit/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "reunit",
```
Comment (yunxing) [inline reUnit/package.json]: Have you thought about publishing this into a separate package?

Comment (bordoley) [inline reUnit/package.json]: Yeah, but I want to work with @glennsl to integrate it better with jest first before encouraging adoption. The idea would be to provide a common DSL that can be used with either a native test runner, or jest for a more full featured experience.

File: release.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  await writeFile (opamFile, opam);
+  await writeFile (metaFile, meta);
+};
+
+const main = async () => {
+  const packageJSON = await readFile (packageJSONFile, "UTF-8");
+  const version = JSON.parse(packageJSON).version;
+
+  await writeFiles (version);
+
```
Comment (tekknolagi) [inline release.js]: Why all this instead of .in conversion?

File: package.json [hunk 5]
```diff
@@ -1,68 +1,26 @@
...
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/facebookincubator/immutable-re.git"
-  },
-  "author": "facebook",
-  "license": "BSD",
-  "homepage": "https://github.com/facebookincubator/immutable-re.git",
   "keywords": [
     "reason",
-    "immutable"
```
Comment (tekknolagi) [inline package.json]: Why remove immutable?

Comment (bordoley) [inline package.json]: i got sloppy.

File: release.js (hunk 4 above)
Comment (bordoley) [inline release.js]: I found for myself that it was easier defining the version in the package.json itself and then generating the META and opam files. I don't think one is necessarily better than the other, but I'm more accustom to JS (npm) workflows.

**Pull request merged**